crates/networking/rpc/engine/payload.rs

```diff
@@ -593,13 +593,6 @@ async fn handle_new_payload_v3(
     block: Block,
     expected_blob_versioned_hashes: Vec<H256>,
 ) -> Result<PayloadStatus, RpcErr> {
-    // Ignore incoming
-    // Check sync status
-    match context.syncer.status()? {
-        SyncStatus::Active(_) => return Ok(PayloadStatus::syncing()),
-        SyncStatus::Inactive => {}
-    }
-
```

Comment:
Removing this check is also quite dangerous, we can execute a payload on top of an invalid state during snap sync without this check

Reply:
We need a way to know if we can execute a payload over a state or not. Do you still store the root node even if the trie is invalid/inconsitent?

Reply:
During snap sync the state is not consistent, we shouldn't execute anything on top of it

Reply:
We can validate the payload (with the detail that missing parent might be due to the sync and not because the block is invalid) but we definitely cannot execute it while snap syncing

Reply:
Geth has logic to delay incoming payloads during an active snap sync for this same reason https://github.com/ethereum/go-ethereum/blob/a7f24c26c0a6a73357d76adf64d98f343bafd0e5/eth/catalyst/api.go#L899

Reply:
Thoughts on this? https://github.com/lambdaclass/ethrex/pull/2426/commits/f0474baa331b5868d6a9e4a43ee2a079a7bb09eb#diff-4fb4a7713efe0d060b54a7903029e99d7d0ecde127b62cfbacaae9344ce827e0R57-R60

or do you prefer to (also) check specifically for snap syncing status?